src/CorrectPlayerMovePredictionPacket.php

```diff
@@ -93,15 +83,9 @@ protected function encodePayload(PacketSerializer $out) : void{
 		$out->putByte($this->predictionType);
 		$out->putVector3($this->position);
 		$out->putVector3($this->delta);
-		if($this->predictionType === self::PREDICTION_TYPE_VEHICLE){
-			if($this->vehicleRotation === null){ // this should never be the case
-				throw new \LogicException("CorrectPlayerMovePredictionPackets with type VEHICLE require a vehicleRotation to be provided");
-			}
-
-			$out->putFloat($this->vehicleRotation->getX());
-			$out->putFloat($this->vehicleRotation->getY());
-			$out->writeOptional($this->vehicleAngularVelocity, $out->putFloat(...));
-		}
+		$out->putFloat($this->vehicleRotation->getX());
+		$out->putFloat($this->vehicleRotation->getY());
+		$out->putFloat($this->vehicleAngularVelocity);
```

Comment:
I haven't checked the spec on this, but this looks weird? Why would they remove the condition on prediction type?

Reply:
Also seems odd to me, but [Cloudburst](https://github.com/CloudburstMC/Protocol/commit/5bb62980960d7519a6c727e2292ca0d55943f290#diff-d5810d2a67587e4f8943d6ba63b5e70d18c14eb1ec0888e333c33dcf75e33da0R19) and the docs seem to say the same thing